python_modules/dagster/dagster/_config/field_utils.py

```diff
@@ -489,14 +490,61 @@ def create(cls, name: str) -> "IntEnvVar":
         var.name = name
         return var
 
+    def __int__(self) -> int:
+        """Raises an exception of the EnvVar value is directly accessed. Users should instead use
+        the `get_value` method, or use the EnvVar as an input to Dagster config or resources.
+        """
+        raise RuntimeError(
+            f'Attempted to directly retrieve environment variable IntEnvVar("{self.name}").'
+            " IntEnvVar should only be used as input to Dagster config or resources."
```

Comment:
Add:

"If you want to access the environment variable directly, you can call `get_value` on this object, or `os.getenv` directly"

Reply:
We could also save.

"The purpose of `EnvVar` is to defer retrieval to right before the step is executed, and to be viewable in the UI"